src/Composer/Compiler.php

```
@@ -142,7 +143,7 @@ public function compile(string $pharFile = 'composer.phar'): void
             __DIR__ . '/../../vendor/symfony/console/Resources/bin/hiddeninput.exe',
             __DIR__ . '/../../vendor/symfony/console/Resources/completion.bash',
         ] as $file) {
-            $extraFiles[$file] = realpath($file);
+            $extraFiles[$file] = Platform::realpath($file);
```

This one should be moved below the file_exists check that happens after realpath..

---

The `RuntimeException` thrown in `Platform::realpath()` doesn't change much:
```
$extraFiles[$file] = realpath($file);
if (!file_exists($file)) {
    throw new \RuntimeException('Extra file listed is missing from the filesystem: '.$file);
}
```
can be replaced with:
```
$extraFiles[$file] = Platform::realpath($file);
```
I'm not sure what the message should be:
>Path does not exist: ...

or
> Could not realpath(): ...

or other.

It can easily be specific to each place `::realpath()` is called, but I think most developers will figure out what's happening from the backtrace rather than the message (assuming it's thrown all the way!).